src/go/rpk/pkg/cli/topic/analyze.go

```diff
@@ -0,0 +1,857 @@
+			}
+		},
+	}
+
+	p.InstallFormatFlag(cmd)
+	cmd.Flags().BoolVarP(&re, "regex", "r", false, "Parse arguments as regex; analyze any topic that matches any input topic expression")
+	cmd.Flags().DurationVar(&timeout, "timeout", 10*time.Second, "Specifies how long the command should run before timing out")
+	cmd.Flags().IntVar(&minBatchesPerPartition, "batches", 10, "Minimum number of batches to consume per partition")
+	cmd.Flags().StringVarP(&timeRng, "time-range", "t", "-1m:end", "Time range to consume from (-24h:end, -48h:-24h, 2022-02-14:1h)")
+
+	cmd.Flags().BoolVarP(&all, "print-all", "a", false, "Print all sections")
+	cmd.Flags().BoolVarP(&pSummary, "print-summary", "s", false, "Print the summary section")
+	cmd.Flags().BoolVar(&pTopics, "print-topics", false, "Print the topics section")
+	cmd.Flags().BoolVar(&pPartitionsBR, "print-partition-batch-rate", false, "Print the detailed partitions batch rate section")
+	cmd.Flags().BoolVar(&pPartitionsBS, "print-partition-batch-size", false, "Print the detailed partitions batch size section")
+
+	return cmd
+}
+
+func validateTopics(ctx context.Context, adm *kadm.Client, topics []string) error {
+	listed, err := adm.ListTopics(ctx, topics...)
+	if err != nil {
+		return err
+	}
+
+	for _, td := range listed {
+		if td.Err != nil {
+			return fmt.Errorf("unable to access topic %q: %v", td.Topic, td.Err.Error())
+		}
+	}
+
+	return nil
+}
+
+type offsetRanges struct {
+	startOffsets kadm.Offsets
+	endOffsets   kadm.Offsets
+}
+
+// For each topic/partition if there is offsets in `tr` then the offsets (so, eo)
+// will be returned where `so` is after `tr.Start` and `eo` is after `tr.End`.
+func getOffsetsForTimeRange(ctx context.Context, adm *kadm.Client, tr timeRange, topics []string) (offsetRanges, error) {
+	var offsets offsetRanges
+
+	lstart, err := adm.ListOffsetsAfterMilli(ctx, tr.Start.UnixMilli(), topics...)
+	if err != nil {
+		return offsets, err
+	}
+	if lstart.Error() != nil {
+		return offsets, fmt.Errorf("unable to list start offsets: %v", lstart.Error())
+	}
+	offsets.startOffsets = lstart.Offsets()
+
+	lend, err := adm.ListOffsetsAfterMilli(ctx, tr.End.UnixMilli(), topics...)
+	if err != nil {
+		return offsets, err
+	}
+	if lend.Error() != nil {
+		return offsets, fmt.Errorf("unable to list end offsets: %v", lend.Error())
+	}
+	offsets.endOffsets = lend.Offsets()
+
+	// Ensure every start offset has an end offset and vice-versa.
+	offsets.startOffsets.KeepFunc(func(so kadm.Offset) bool {
+		_, exists := offsets.endOffsets.Lookup(so.Topic, so.Partition)
+		return exists
+	})
+	offsets.endOffsets.KeepFunc(func(so kadm.Offset) bool {
+		_, exists := offsets.startOffsets.Lookup(so.Topic, so.Partition)
+		return exists
+	})
+
+	return offsets, nil
+}
+
+// Used to allow for time.Time to be printed correctly in '--format help'.
+type Time struct {
+	time.Time
+}
+
+// MarshalText implements encoding.TextMarshaler.
+func (t Time) MarshalText() ([]byte, error) {
```

Comment:
```suggestion
func (t *Time) MarshalText() ([]byte, error) {
```

Reply:
I was mainly copying how golang's `time.Time` was implementing these. And oddly enough it is using a value receiver for `MarshalText`. My guess is that `time.Time` is likely close in size to a pointer anyhow so adding further indirection doesn't gain us anything over just making a copy.

Happy to change this though if you feel strongly about the matter.